src/Mercure/EventListener/AddLinkHeaderListener.php

```diff
@@ -43,26 +47,34 @@ public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFa
     public function onKernelResponse(ResponseEvent $event): void
     {
         $request = $event->getRequest();
-        // Prevent issues with NelmioCorsBundle
         if ($this->isPreflightRequest($request)) {
             return;
         }
 
-        $link = new Link('mercure', $this->hub);
-
         if (
             null === ($resourceClass = $request->attributes->get('_api_resource_class')) ||
-            false === $this->resourceMetadataFactory->create($resourceClass)->getAttribute('mercure', false)
+            false === ($mercure = $this->resourceMetadataFactory->create($resourceClass)->getAttribute('mercure', false))
         ) {
             return;
         }
 
-        if (null === $linkProvider = $request->attributes->get('_links')) {
-            $request->attributes->set('_links', new GenericLinkProvider([$link]));
+        if (!$this->discovery instanceof Discovery) {
+            @trigger_error(sprintf('Passing a string as the second argument to "%s::__construct()" is deprecated, pass a "%s" instance instead.', __CLASS__, Discovery::class), \E_USER_DEPRECATED);
```

Comment:
wdyt about beeing more specific on this message mentioning that we switched from a string to a Discovery class?

Reply:
Yep sounds good! any suggestions?

Reply:
Using a string for the Mercure hub in the constructor of __CLASS__ is deprecated since 2.6 in favor of an instance of Discovery::class.

Reply:
It's perfect! Thanks